pkg/proxy/iptables/proxier_test.go

```diff
@@ -3575,3 +3581,554 @@ COMMIT
 		}
 	}
 }
+
```

Comment:
we have to test

>  In addition, kube-proxy's node port health check should fail if there are only Terminating endpoints, regardless of their readiness in order to:

Reply:
Good catch, let me add one

Reply:
Added more tests for this, see last 3 commits